### ISSUE TITLE
DHCP doesn't always answer to routed networks

### ISSUE THREAD

**Issue by julsemaan** [inverse-inc/packetfence#3407]
In the case where the gateway IP (router of the routed network) is not the source IP of the packet when its being relayed, and that this source IP isn't being routed through the source interface, then pfdhcp doesn't answer to packets.

The gateway IP in the DHCP packet itself is good, so when dealing with L3 networks, this should have precedence over the source IP of the packet

**Comment by julsemaan** [comment]
TL;DR
Add this to sysctl.conf
```
net.ipv4.conf.all.rp_filter = 2
net.ipv4.conf.default.rp_filter = 2
```

It ended up being the DHCP relays using the management interface of the routed network as the relay source IP address (in the IP frame) instead of the interface of the routed network.

The gateway IP was correct in the DHCP packet but since the packet wasn't entering the right interface (entered on the registration interface while the route for the packet was on the management interface), then the packet was left unanswered.

We confirmed that ISC DHCP doesn't exhibit this behavior and manages to get the packets.
Only thing is that as @fdurand discovered, the packet never enters the pfdhcp service on the port bind.
We would have to see what ISC DHCP does, perhaps using packet capture instead of the port bind for handling the inbound packets

**Comment by fdurand** [comment]
IMO even if ISC DHCP is able to handle that because it capture dhcp packet instead of opening a socket for me it's a network configuration issue and this should be fixed on the network side.
What we can do is to detect this network miss-configuration and alert the network admin to do the correct configuration.

**Comment by julsemaan** [comment]
Would be good enough to have the alert but I'm unsure how pfdhcp can detect it since its not getting the packets. Unless we start a pcap just for this purpose but it seems a bit overkill or as @lzammit would say "tuer des mouches à la mitraillette"

Unless you have a better idea to perform that detection.

**Comment by fdurand** [comment]
We have the pfdhcplistener for that.
